one-pager-maker/src/pages/SetId.tsx

```diff
@@ -0,0 +1,92 @@
+import { useCallback, useState } from 'react';
+import {
+    Box,
+    Button,
+    TextField,
+} from '@mui/material';
+import { useNavigate } from 'react-router-dom';
+import { userApi } from "../api/userApi.ts";
+import { assertZodSchema } from "../utils/asserts.ts";
+import { idSchema } from '../entity/user/userType.ts';
+import { useAppSelector } from '../redux/hooks.ts';
+
+const SetId = () => {
+
+    const navigate = useNavigate();
+
+    const [formData, setFormData] = useState({ userId: "" });
+
+    const uid = useAppSelector(state => state.user.user?.uid);
+    const photoUrl = useAppSelector(state => state.user.user?.photoUrl);
+    const result = userApi.useIsDuplicatedIdQuery(formData.userId);
+    const mutation = userApi.useCreateUserMutation();
+
+    const handleFormData = useCallback((event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+        setFormData(formData => ({
+            ...formData,
+            [event.target.id]: event.target.value,
+        }))
+    }, [])
+
+    const updateUser = useCallback(() => {
+
+        // user id validation
+        try {
+            assertZodSchema(idSchema, formData.userId)
+        } catch {
+            alert('User ID is allowed only alphanumeric characters, underscores (_), and hyphens (-).');
+            return
+        }
```

Comment:
Zod has safeParse() and this allows to remove try-catch statement.
Use it if you want
```TYpeScript
const result = idSchema.safeParse(formData.userId);

if (!result.success) {
  // if failed
}
```

Reply:
Thanks. I fix it.